### ISSUE TITLE
Fail to install pywbem-0.8.2 on Windows

### ISSUE THREAD

**Issue by zhaoy8** [pywbem/pywbem#219]
On Windows, installing **pywbem-0.8.2** from pypi fails due to **m2crypto** compiling issue.

**Comment by andy-maier** [comment]
Hi.
How exactly does it fail, can you provide the messages that are displayed?
Andy

**Comment by jealous** [comment]
See the full error log here.
[error.txt](https://github.com/pywbem/pywbem/files/218871/error.txt)

By the way, I already have m2crypto installed.

```
c:\work>pip show m2crypto
---
Metadata-Version: 1.0
Name: m2crypto
Version: 0.21.1
Summary: M2Crypto: A Python crypto and SSL toolkit
Home-page: http://chandlerproject.org/Projects/MeTooCrypto
Author: Heikki Toivonen
Author-email: heikki@osafoundation.org
License: BSD-style license
Location: c:\tools\anaconda\lib\site-packages
Requires:
Classifiers:
  Development Status :: 5 - Production/Stable
  Intended Audience :: Developers
  Operating System :: OS Independent
  Programming Language :: C
  Programming Language :: Python
  Topic :: Security :: Cryptography
  Topic :: Software Development :: Libraries :: Python Modules
```

**Comment by andy-maier** [comment]
The error happens when the pywbem setup.py script attempts to build the LEX/YACC table modules. During that process, the pywbem package is imported, which in turn imports the M2Crypto package, and this is where it fails.

This can happen in v0.8.2 when M2Crypto has been installed in the same install run, but it may depend on the Python implementation (you are using Anaconda), the platform, the setup.py command, etc.

We improved this particular area of the installation of pywbem a bit in the upcoming 0.8.3 version. I added some more debug and error analysis code in case it fails. The result is in PR #220.

-> Could you please retry with that version?

To do that, download and unpack the [pywbem distribution archive](https://github.com/pywbem/pywbem/blob/fix_issue_219/dist/pywbem-0.8/pywbem-0.8.3.dev0.tar.gz), and install from there, using:

    python setup.py install

and post the result here.

**Comment by andy-maier** [comment]
If that works, I'll release v0.8.3 to PyPI and you can retry with your normal installation procedure.

**Comment by Napsty** [comment]
By the way the same also happens on Linux Mint 17.3 (Ubuntu 14.04)

    # cat /etc/linuxmint/info 
    RELEASE=17.3
    CODENAME=rosa
    EDITION="Cinnamon 64-bit"
    DESCRIPTION="Linux Mint 17.3 Rosa"
    DESKTOP=Gnome
    TOOLKIT=GTK
    NEW_FEATURES_URL=http://www.linuxmint.com/rel_rosa_cinnamon_whatsnew.php
    RELEASE_NOTES_URL=http://www.linuxmint.com/rel_rosa_cinnamon.php
    USER_GUIDE_URL=help:linuxmint
    GRUB_TITLE=Linux Mint 17.3 Cinnamon 64-bit

Installation with pip fails with the same error:

	# pip install pywbem
	Downloading/unpacking pywbem
	  Downloading pywbem-0.8.2.tar.gz (180kB): 180kB downloaded
	  Running setup.py (path:/tmp/pip_build_root/pywbem/setup.py) egg_info for package pywbem
		
	Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from pywbem)
	Downloading/unpacking ply (from pywbem)
	  Downloading ply-3.8.tar.gz (157kB): 157kB downloaded
	  Running setup.py (path:/tmp/pip_build_root/ply/setup.py) egg_info for package ply
		
		warning: no previously-included files matching '*.pyc' found anywhere in distribution
	Downloading/unpacking M2Crypto (from pywbem)
	  Downloading M2Crypto-0.24.0.tar.gz (184kB): 184kB downloaded
	  Running setup.py (path:/tmp/pip_build_root/M2Crypto/setup.py) egg_info for package M2Crypto
		
		warning: no previously-included files found matching 'M2Crypto/_m2crypto.py'
	Installing collected packages: pywbem, ply, M2Crypto
	  Running setup.py install for pywbem
		usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
		   or: -c --help [cmd1 cmd2 ...]
		   or: -c --help-commands
		   or: -c cmd --help
		
		error: option --single-version-externally-managed not recognized
		Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/pywbem/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-q11Xsb-record/install-record.txt --single-version-externally-managed --compile:
		usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]

	   or: -c --help [cmd1 cmd2 ...]

	   or: -c --help-commands

	   or: -c cmd --help



	error: option --single-version-externally-managed not recognized

**Comment by Napsty** [comment]
Manual installation of your mentioned 0.8.3 (https://github.com/pywbem/pywbem/blob/fix_issue_219/dist/pywbem-0.8/pywbem-0.8.3.dev0.tar.gz) worked fine.
Output: https://gist.github.com/Napsty/ef65351f028bd36aa6662c4c41aa551a

**Comment by andy-maier** [comment]
Denny, Cedric,
I updated the the [pywbem distribution archive](https://github.com/pywbem/pywbem/blob/fix_issue_219/dist/pywbem-0.8/pywbem-0.8.3.dev0.tar.gz) for this branch, to add Windows batch file versions of the mof_compiler and wbemcli scripts, and to use M2CryptoWin32/64 when on Windows, instead of M2Crypto.

Please try that version on Windows, to see whether it works with your Anaconda Python.

**Comment by jealous** [comment]
[8.3.dev.txt](https://github.com/pywbem/pywbem/files/220255/8.3.dev.txt)
Thank you for the quick fix, Andy!
It works fine in windows now.
Check the log file for detail.

**Comment by andy-maier** [comment]
Very good. I'll release 0.8.3 then.

**Comment by zhaoy8** [comment]
I also verified 0.8.3 on my Windows machine. Thanks Andy!

**Comment by andy-maier** [comment]
PR #220 which fixes this issue has been merged into the v0.8.x stream (stable_0.8).
v0.8.3 will be released soon.

**Comment by andy-maier** [comment]
v0.8.3 has been released: https://github.com/pywbem/pywbem/releases/tag/v0.8.3
Closing this issue.
